lib/elixir_analyzer/test_suite/strain.ex

```diff
@@ -0,0 +1,37 @@
+defmodule ElixirAnalyzer.TestSuite.Strain do
+  @moduledoc """
+  This is an exercise analyzer extension module for the concept exercise Strain
+  """
+
+  use ElixirAnalyzer.ExerciseTest
+
+  assert_no_call "does not call Enum.filter/2" do
+    type :essential
+    called_fn module: Enum, name: :filter
+    comment ElixirAnalyzer.Constants.strain_use_recursion()
+  end
+
+  assert_no_call "does not call Enum.reject/2" do
+    type :essential
+    called_fn module: Enum, name: :reject
+    comment ElixirAnalyzer.Constants.strain_use_recursion()
+  end
```

Comment:
Those functions could be recreated with `Enum.reduce` :(. If `strain` is really a recursion-only exercise, then we need to forbid the whole `Enum`, `List`, `Stream` modules and list comprehensions. Like for `bird-count`.

Reply:
Allowing only a single `Enum` function might be confusing 🤔 I guess the question is what we want from this exercise. If it's supposed to be an exercise of recursion, I guess `Enum.reverse` should not be allowed either and the example should be updated. @jiegillet @neenjaw opinions?

The most starred solution uses simple recursion (no accumulator, no reverse): https://exercism.org/tracks/elixir/exercises/strain/solutions/morgangrubb

Reply:
Whenever the exercise is "recreate an `Enum` function", solutions I like are always either `reduce` + `reverse` (aka `foldr`) or plain recursion.
My own solution is very similar to the one you link, plain recursion. However I think I would not keep people away from using other `Enum` functions. So I vote to only reject `filter` and `reject`.

Reply:
Ok then, let's keep this check like it is now. We just need the comment for the PR to be ready :)